wandb/sdk/artifacts/artifact.py

```diff
@@ -1814,6 +1819,38 @@ def _download_entry(
             )
         return FilePathStr(root)
 
+    def _restore_from_cache(
+        self, root: str, path_prefix: Optional[StrPath] = None
+    ) -> bool:
+        """Restore all files available in the cache to the specified root directory.
+
+        Returns:
+            True if all files were restored, False if any entries are missing.
+        """
+        cache = get_artifact_file_cache()
+
+        def restore_entry(entry):
```

Comment:
hmm how fast are these checks? Do we have the opposite problem where for super large artifacts/references, this introduces non-trivial overhead before we commence downloading?